csharp/src/Google.Protobuf/Reflection/DescriptorPool.cs

```diff
@@ -189,13 +186,38 @@ private static void ValidateSymbolName(IDescriptor descriptor)
             {
                 throw new DescriptorValidationException(descriptor, "Missing name.");
             }
-            if (!ValidationRegex.IsMatch(descriptor.Name))
+
+            if (!IsValidSymbolNameFormat(descriptor.Name))
             {
                 throw new DescriptorValidationException(descriptor,
                                                         "\"" + descriptor.Name + "\" is not a valid identifier.");
             }
         }
 
+        /// <summary>
+        /// Symbol name must start with a letter or underscore, and it can contain letters, numbers and underscores.
+        /// </summary>
+        private static bool IsValidSymbolNameFormat(string name)
+        {
+            if (!IsAsciiLetter(name[0]) && name[0] != '_')
```

Comment:
Possibly add a comment indicating that an empty name is already checked before the call to this method?
I'd also probably make it a local method to ValidateSymbolName - otherwise it *could* be called by other code in this class in the future with an empty string.

Reply:
Refactored to remove this method.